src/base/QXmppHttpUploadIq.h

```diff
@@ -44,6 +45,8 @@ class QXMPP_EXPORT QXmppHttpUploadRequestIq : public QXmppIq
     QXmppHttpUploadRequestIq();
     ~QXmppHttpUploadRequestIq() override;
 
+    QXmppHttpUploadRequestIq &operator=(const QXmppHttpUploadRequestIq &);
```

Comment:
Adding move ctors and move assignment operators might be a good idea for this and similar classes

Reply:
Right. Just what I'm not sure about are QObject-based classes. Does it even make sense to use QSharedData there? - QObject-based classes cannot be copied usually.

Reply:
I added missing move ctors to the MIX item classes and HTTP Upload IQs.

I think I should better keep QXmppStreamPrivate and the RTP channel classes as they are - the only advantage of QSharedData here is that you don't need to manually delete the objects in the end.